File: Algorithm/NumberOfRectanglesThatCanFormTheLargestSquare.cpp
class Solution {
public:
    int countGoodRectangles(vector<vector<int>>& rectangles) {
        int cur = 0;
        int ans = 0;
        for (const auto& r : rectangles) {
            if (min(r[0], r[1]) > cur) {
                cur = min(r[0], r[1]);
                ans = 0;
            }
            if (min(r[0], r[1]) == cur) ++ans;
        }
        return ans;
    }
};

File: Algorithm/NumberOfRectanglesThatCanFormTheLargestSquare.java
// Solution: Running Max of Shortest Edge
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
    public int countGoodRectangles(int[][] rectangles) {
        int cur = 0;
        int ans = 0;
        for (int[] r : rectangles) {
            if (Math.min(r[0], r[1]) > cur) {
                cur = Math.min(r[0], r[1]);
                ans = 0;
            }
            if (Math.min(r[0], r[1]) == cur) ++ans;
        }
        return ans;
    }
}
